### PR TITLE
Initial support for Xilinx Versal

### PR DESCRIPTION
**Pull request by ricardosalveti**

### CODE REVIEW

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 1]
```diff
@@ -498,6 +498,25 @@ SOTA_EXTRA_CLIENT_FEATURES:append:sota:uz3eg-iocc-sec = " fiovb"
...
+PREFERRED_PROVIDER_u-boot-default-script:sota:versal = "u-boot-ostree-scr-fit"
+PREFERRED_PROVIDER_virtual/kernel:versal = "linux-lmp-xlnx"
+PREFERRED_PROVIDER_virtual/dtb:versal = "device-tree"
+PREFERRED_PROVIDER_virtual/trusted-firmware-a:versal = "arm-trusted-firmware"
+KERNEL_BUILTIN_WIREGUARD:versal = "1"
+EXTRA_IMAGEDEPENDS:remove:versal = "qemu-devicetrees"
+UBOOT_ENTRYPOINT:versal = "0x18000000"
+OSTREE_KERNEL_ARGS:versal ?= "console=ttyAMA0,115200 ${OSTREE_KERNEL_ARGS_COMMON}"
+KERNEL_IMAGETYPE:sota:versal = "fitImage"
+KERNEL_CLASSES:sota:versal = " kernel-lmp-fitimage "
```
Comment (angolini) [inline meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:517]: is this line missing an `:append` or the extra space are only extra? (just double checking)

File: meta-lmp-base/recipes-kernel/linux/kmeta-linux-lmp-5.10.y.inc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-KERNEL_META_REPO ?= "git://github.com/foundriesio/lmp-kernel-cache.git"
+KERNEL_META_REPO ?= "git://github.com/ricardosalveti/lmp-kernel-cache.git"
```
Comment (angolini) [inline meta-lmp-base/recipes-kernel/linux/kmeta-linux-lmp-5.10.y.inc:1]: just saw it's a draft.....

**Review by angolini — COMMENTED** [review]
...

**Comment by ldts** [comment]
# Xilinx
USE_XSCT_TARBALL_zynqmp = "1"

I think we need the same for Versal...

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:517]: Don't think we need the extra space, just a copy & paste issue.

**Comment by ricardosalveti** [comment]
> USE_XSCT_TARBALL_zynqmp = "1"
> 
> I think we need the same for Versal...

Indeed, added.

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc]: Removed.

**Comment by ricardosalveti** [comment]
@angolini  should be ready for a review now.

**Review by angolini — APPROVED** [review]
LGTM

**Comment by ricardosalveti** [comment]
@ldts will merge but changes are still expected in order to get a fully working build in place, this is to unblock from the build perspective.

**Pull request merged**